src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1403,6 +1403,10 @@ private BoundExpression GetDefaultParameterValue(SyntaxNode syntax, ParameterSym
                 // so replace the argument with default(int?).
                 defaultValue = new BoundDefaultExpression(syntax, parameterType) { WasCompilerGenerated = true };
             }
+            else if (defaultConstantValue.IsNull || defaultConstantValue.IsBad)
```

Comment:
In the `void M(MyStruct? s = default(MyStruct))` scenario, the `defaultConstantValue.IsBad` value is true. This change prevents us from entering the `parameterType.IsNullableType()` branch.

The original code was probably already preventing other kinds of bad default arguments from crashing the compiler--it just hadn't accounted for bad arguments to System.Nullable parameters.